web/src/components/wizard/SetupStep.tsx

```diff
@@ -16,57 +17,98 @@ const SetupStep: React.FC<SetupStepProps> = ({ onNext, onBack }) => {
+  const { isLoading: isConfigLoading } = useQuery({
+    queryKey: ["installConfig"],
+    queryFn: async () => {
+      const response = await fetch("/api/install", {
+        headers: {
+          ...(localStorage.getItem("auth") && {
+            Authorization: `Bearer ${localStorage.getItem("auth")}`,
+          }),
+        },
+      });
+      if (!response.ok) {
+        throw new Error("Failed to fetch install configuration");
       }
-    };
+      const data = await response.json();
+      updateConfig(data.config);
+      return data;
+    },
+  });
+
+  // Query for fetching network interfaces
+  const { data: networkInterfacesData, isLoading: isInterfacesLoading } =
+    useQuery({
+      queryKey: ["networkInterfaces"],
+      queryFn: async () => {
+        const response = await fetch(
+          "/api/console/available-network-interfaces",
+          {
+            headers: {
+              ...(localStorage.getItem("auth") && {
+                Authorization: `Bearer ${localStorage.getItem("auth")}`,
+              }),
+            },
+          }
+        );
+        if (!response.ok) {
+          throw new Error("Failed to fetch network interfaces");
+        }
+        return response.json();
+      },
+    });
+
+  // Mutation for submitting the configuration
+  const {
+    mutate: submitConfig,
+    isPending: isSubmitting,
+    error: submitError,
+  } = useMutation({
+    mutationFn: async (configData: typeof config) => {
+      const response = await fetch("/api/install/config", {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+          ...(localStorage.getItem("auth") && {
+            Authorization: `Bearer ${localStorage.getItem("auth")}`,
+          }),
+        },
+        body: JSON.stringify(configData),
+      });
 
-    fetchData();
-  // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, []);
+      if (!response.ok) {
+        const errorData = await response.json().catch(() => ({}));
+        throw errorData;
+      }
+      return response.json();
+    },
+    onSuccess: () => {
+      onNext();
+    },
+    onError: (err: any) => {
+      setError(err.message || "Failed to setup cluster");
+    },
+  });
+
+  // Helper function to get field error message
+  const getFieldError = (fieldName: string) => {
+    if (!submitError?.errors) return undefined;
+    const fieldError = submitError.errors.find(
+      (err: any) => err.field === fieldName
+    );
+    return fieldError?.message;
+  };
 
   const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
     const { id, value } = e.target;
-    if (id === 'adminConsolePort') {
-      updateConfig({ adminConsolePort: parseInt(value) });
-    } else if (id === 'localArtifactMirrorPort') {
-      updateConfig({ localArtifactMirrorPort: parseInt(value) });
+    if (id === "adminConsolePort" || id === "localArtifactMirrorPort") {
+      // Only update if the value is empty or a valid number
```

Comment:
does this mean that if they type anything else it wouldn't show anything being typed?